onnx/funconnx/proto_run.py

```diff
@@ -0,0 +1,368 @@
+        results = sess.run(None, {"X": X})
+        print(results[0])  # display the first result
+
+    The class can be use any implementation available in folder
+    `aionnx <https://github.com/onnx/onnx/tree/main/onnx/runtime/aionnx>`_.
+    Adding an implementation requires two changes. The first is
+    the implementation itself. Any existing node can be used as a template.
+    The second is one line in file `_op_list.py
+    <https://github.com/onnx/onnx/tree/main/onnx/runtime/aionnx/_op_file.py>`_
+    to import the file and let the runtime know it exists.
+
+    This class can also be used to test an implementation of
+    a custom operator. Let's assume this new operator
+    is `InvAlpha` from domain `custom`. The implementation
+    must take place in a class inheriting from
+    :class:`OpRun <onnx.funconnx.op_run.OpRun>`.
+    It must also define attribute `op_domain`.
+    Here is an example which computes :math:`\\frac{1}{X + \\alpha}`.
+
+    .. exec_code::
+
+        from onnx.funconnx.op_run import OpRun
+
+        class InvAlpha(OpRun):
+
+            op_domain = "custom"
+
+            def _run(self, x, alpha=None):  # type: ignore
+                # None must be the default value, it is automatically
+                # replaced by class OpRun with either the default value
+                # defined by ONNX or the value specified in the NodeProto.
+                return (1 / (x + alpha),)
+
+    `alpha` is an attribute. It can be defined by the onnx node or
+    be defined by the function used this node. It is possible
+    Line `alpha = alpha or self.alpha` selects first the value defined
+    the onnx function if that's the case of falls back to the default
+    value defined by the onnx node.
+    to link a function attribute to a node attribute. In that case,
+    Class `ProtoRun` must know about this new implementation
+    and this can be done by specified argument *new_ops*.
+
+    ::
+
+        sess = ProtoRun(onnx_model, new_ops=[InvAlpha])
+        got = sess.run(None, {"X": x})[0]
+
+    A specific node can be simply evaluated.
+
+    .. exec_code::
+
+        import numpy as np
+        from onnx.funconnx.aionnx._op_list import Celu
+
+        x = np.array([[0, 1], [-1, 2]], dtype=np.float32)
+        y = Celu.eval(x, alpha=0.5)
+        print(y)
+
+    This can also be expressed as:
+
+    .. exec_code::
+
+        import numpy as np
+        from onnx.funconnx.aionnx import load_op
+
+        Celu = load_op("", "Celu")  # domain is ""
+        x = np.array([[0, 1], [-1, 2]], dtype=np.float32)
+        y = Celu.eval(x, alpha=0.5)
+        print(y)
+
+    Method :meth:`eval <onnx.ProtoRun.op_run.eval>` creates an onnx node
+    returned by method :meth:`make_node <onnx.funconnx.op_run.make_node>`.
+
+    .. exec_code::
+
+        import numpy as np
+        from onnx.funconnx.aionnx._op_list import Celu
+
+        onnx_node = Celu.make_node(alpha=0.5)
+        print(onnx_node)
```

Comment:
nit: The above few lines (113-122) seem irrelevant here and can be omitted. In particular, it is an implementation detail of `eval`, and I think we can change `eval` to work without constructing a node, so not worth describing those details here.

Reply:
moved into the docstring of make_node.